src/dune_rpc_impl/server.ml

```diff
@@ -230,13 +233,18 @@ let handler (t : t Fdecl.t) : 'a Dune_rpc_server.Handler.t =
     t.clients <- Clients.add_session t.clients session;
     Fiber.return client
   in
+  let on_upgrade session menu =
+    let+ () = Fiber.return () in
```

Comment:
Is this necessary because this introduces a delay (by yielding to the scheduler) or simply because we need to wrap the result into a `Fiber.t`? 

(I hope it's not for the delay, so maybe we can rewrite the code to make it a bit more obvious or add a comment.)

Reply:
> by yielding to the scheduler

Note that it does no such thing. Yielding to the scheduler only happens when calling `Fiber.Ivar.read` on an unfilled ivar.

> or simply because we need to wrap the result into a Fiber.t?

That's the only reason indeed.

Reply:
Ah, indeed, I confused things up. Thanks!